crates/utils/src/version.rs

```diff
@@ -0,0 +1,21 @@
+//! Utilities for reading version number
+
+use hotshot_constants::Version;
+
+/// Read the version number from a message (passed a byte vector),
+/// returning None if there are not enough bytes.
+#[must_use]
+#[allow(clippy::module_name_repetitions)]
+pub fn read_version(message: &[u8]) -> Option<Version> {
+    let maybe_bytes_major: Result<[u8; 2], _> = message[0..2].try_into();
+    let maybe_bytes_minor: Result<[u8; 2], _> = message[2..4].try_into();
```

Comment:
Looks like this will panic if the length of `message` is less than 4. I think we can do a length check before this.

Reply:
Oops, yes! I missed that taking the slice can panic.

I rewrote this with `get` instead, and added some unit tests to make sure it really does return `None` (and doesn't panic) if the message is too short.